packages/model-instance-client/src/events.ts

```diff
@@ -186,8 +196,13 @@ export async function createDataEvent<
   )
   // Header must only be provided if there are values
   // CBOR encoding doesn't support undefined values
-  const header =
-    params.shouldIndex == null ? undefined : { shouldIndex: params.shouldIndex }
+  const header: DocumentDataEventHeader = {}
```

Comment:
Does it matter that we were sending an `undefined` header before if `shouldIndex == null`? We'll always pass the header into the function now, and it might be an empty map.

Reply:
Or should we just update `createDataEventPayload` to ignore both an undefined `header` and an empty one?

Reply:
Or does it not matter if `header` is empty?

Reply:
Good call, I am 90% sure it doesn't matter but better safe than sorry. I'll change this to not set a header if shouldIndex or modelVersion are not set.